src/pbt/prophecy_build_tool.py

```diff
@@ -0,0 +1,472 @@
+                    is_shell=True,
+                ),
+                # Run the unit test
+                Process(
+                    ["python3", "-m", "pytest", "-v", "test/TestSuite.py"],
+                    path_pipeline_absolute,
+                ),
+            ]
+        )
+
+    def _verify_unit_test_env(self):
+        self.fabric = os.environ.get("FABRIC_NAME")
+
+        if self.fabric is None:
+            self._error(
+                "[i]FABRIC_NAME[/i] environment variable is required to "
+                "run your Unit tests"
+            )
+            return False
+        else:
+            return True
+
+    def _parse_project(self):
+        self.pipelines: Dict = {}
+        self.jobs: Dict = {}
+        with open(self.path_project, "r") as _in:
+            self.project = yaml.safe_load(_in)
+            self.jobs = self.project["jobs"]
+            self.pipelines = self.project["pipelines"]
+            self.project_language = self.project["language"]
+
+        if self.project_language not in ("python", "scala"):
+            self._error(
+                f"Language: [i]{self.project_language}[/i] not supported by Prophecy-Build-Tool right now."
+            )
+
+        self.pipelines_count = len(self.pipelines)
+        self.jobs_count = len(self.jobs)
+
+        jobs_str = ", ".join(map(lambda job: job["name"], self.jobs.values()))
+        print("Found %s jobs: %s" % (self.jobs_count, jobs_str))
+
+        pipelines_str = ", ".join(
+            map(
+                lambda pipeline: "%s (%s)" % (pipeline["name"], pipeline["language"]),
+                self.pipelines.values(),
+            )
+        )
+        print("Found %s pipelines: %s" % (self.pipelines_count, pipelines_str))
+        self._verify_project_structure()
+
+    def _verify_project_structure(self):
+        for path_pipeline, pipeline in self.pipelines.items():
+            path_pipeline_absolute = os.path.join(
+                os.path.join(self.path_root, path_pipeline), "code"
+            )
+            pipeline_dependencies_file = (
+                "setup.py" if self.project_language == "python" else "pom.xml"
+            )
+
+            if not os.path.isfile(
+                os.path.join(path_pipeline_absolute, pipeline_dependencies_file)
+            ):
+                print(
+                    f"\n[bold red]❌ Pipeline {path_pipeline} does not exist or is corrupted. [/bold red]"
+                )
+                sys.exit(1)
+
+        for path_job, job in self.jobs.items():
+            path_job_definition = os.path.join(
+                os.path.join(self.path_root, path_job), "code/databricks-job.json"
+            )
+
+            if not os.path.isfile(path_job_definition):
+                print(
+                    f"\n[bold red]❌ Job {path_job} does not exist or is corrupted. [/bold red]"
+                )
+                sys.exit(1)
+
+    def _process_sequential(self, processes, time_between_each_cmd=1):
```

Comment:
Please make this method static. Doesn't matter a lot now but becomes a good practice as this beautiful build tool grows

Reply:
Done